core/Pointer.carp

```diff
@@ -20,6 +20,28 @@ The user will have to ensure themselves that this is a safe operation.")
   (doc from-long "converts a long integer to a pointer.")
   (deftemplate from-long (Fn [Long] (Ptr p)) "$p* $NAME(Long p)" " $DECL { return ($p*)p; }")
 
+  (doc unsafe-set
+    "Sets the value of a pointer to a value of any type."
+    "The user will have to ensure this operation is safe.")
+  (deftemplate unsafe-set (Fn [(Ptr p) a] Unit) "void $NAME($p* p, $a a)" "$DECL { *p = ($p)a; }")
+
+  (doc unsafe-alloc
+     "Allocate a new pointer to the value `a`."
+     ("This pointer won't be managed by Carp's borrow checker and will cause leaks " false)
+     "unless explicitly freed."
+     ""
+     ("See `Unsafe.leak` if you want to prevent Carp's automatic destructuring " false)
+     "on a value without performing an allocation.")
```

Comment:
What purpose do the `("doc string" false)` bits serve?

Maybe nitpicking but I would replace `destructuring` with `memory management`, for me destructuring is splitting a value into its component parts.

Reply:
the `false` arg is a newer bit introduced in #935 -- basically `doc` can take a rest parameter of strings now instead of one big string--each string is newline separated by default, but an optional `false` arg turns off the newline. It's just to enable readability in the source.

You are completely correct on `destructing`! I'll update the doc